wallet/src/slip32.rs

```diff
@@ -0,0 +1,68 @@
+//! Slip32 implementation
+
+use crate::key;
+
+/// Byte sizes for each field used in the serialization format
+// pub struct SerializationFormat {
+//     version: u8,
+//     depth: u8,
+//     fingerpint: u8,
+//     child_number: u8,
+//     chain_code: u8,
+//     key: u8
+// }
+
+/// Export a BIP32 extended key to the SLIP32 format
+pub fn export<K>(path: key::KeyPath, extkey: key::ExtendedKey<K>) -> String
+where
+    K: key::Key,
+{
+    let depth = path.depth();
+    let mut buffer = Vec::with_capacity(buffer_size(depth));
+
+    // 1 byte for depth
+    buffer[0] = depth as u8;
+
+    // 4 bytes for each child number in the derivation path
+    for (level, child_number) in path.iter().enumerate() {
+        let start = 1 + level * 4;
+        let end = start + 4;
+        buffer.splice(start..end, child_number.to_bytes().iter().cloned());
+    }
+
+    // 32 bytes for chain code
+    {
+        let start = 1 + depth * 4;
+        let end = start + 32;
+        buffer.splice(start..end, extkey.chain_code.iter().cloned());
+    }
+
+    // 33 bytes for key
+    //
+    // TODO: Find a way to get access to secp256k1::SecretKey bytes,
```

Comment:
In https://github.com/rust-bitcoin/rust-secp256k1/issues/75#issuecomment-456607681 they answer how to get the bytes out from a secret key. 

> You should be able to use [..] to get the bytes out of a secret key no problem. See, eg, this line in rust-lightning: https://github.com/rust-bitcoin/rust-lightning/blob/master/src/ln/chan_utils.rs#L80